test/framework/kogitobuild.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/resource"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	"github.com/kiegroup/kogito-cloud-operator/test/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeployKogitoBuild deploy a KogitoBuild
+func DeployKogitoBuild(namespace string, installerType InstallerType, kogitoBuild *v1alpha1.KogitoBuild) error {
+	GetLogger(namespace).Infof("%s deploy %s example %s with name %s and native %v", installerType, kogitoBuild.Spec.Runtime, kogitoBuild.Spec.GitSource.ContextDir, kogitoBuild.Name, kogitoBuild.Spec.Native)
+
+	switch installerType {
+	case CRInstallerType:
+		return crDeployKogitoBuild(namespace, kogitoBuild)
+	default:
+		panic(fmt.Errorf("Unknown installer type %s", installerType))
+	}
+}
+
+func crDeployKogitoBuild(namespace string, kogitoBuild *v1alpha1.KogitoBuild) error {
+	if _, err := kubernetes.ResourceC(kubeClient).CreateIfNotExists(kogitoBuild); err != nil {
+		return fmt.Errorf("Error creating example service %s: %v", kogitoBuild.Name, err)
+	}
+	return nil
+}
+
+// GetKogitoBuildStub Get basic KogitoBuild stub with all needed fields initialized
+func GetKogitoBuildStub(namespace, runtimeType, name string) *v1alpha1.KogitoBuild {
+	kogitoBuild := &v1alpha1.KogitoBuild{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Status: v1alpha1.KogitoBuildStatus{
+			Conditions: []v1alpha1.KogitoBuildConditions{},
+		},
+		Spec: v1alpha1.KogitoBuildSpec{
+			Runtime:        v1alpha1.RuntimeType(runtimeType),
+			MavenMirrorURL: config.GetMavenMirrorURL(),
+		},
+	}
+
+	return kogitoBuild
+}
+
+// SetupKogitoBuildImageStreams sets the correct images for the KogitoBuild
+func SetupKogitoBuildImageStreams(kogitoBuild *v1alpha1.KogitoBuild) {
+	kogitoBuild.Spec.BuildImage = getKogitoBuildS2IImage(kogitoBuild)
+	kogitoBuild.Spec.RuntimeImage = getKogitoBuildRuntimeImage(kogitoBuild)
+}
+
+func getKogitoBuildS2IImage(kogitoBuild *v1alpha1.KogitoBuild) v1alpha1.Image {
```

Comment:
This method is almost duplicated in [here](https://github.com/kiegroup/kogito-cloud-operator/blob/2a15042e1717c1af2f6d0e8b08803b69d01f316f/test/framework/kogitoapp.go#L216). Maybe, we could refactor it to make use of it also here.

Reply:
Yes, you are right.
But once KogitoApp is removed then the referenced implementation is removed too and this one will become the only one.
So I just didn't want to refactor something which will be removed in future anyway.

Reply:
Ok, it makes sense.